pipe.go

```diff
@@ -159,6 +184,10 @@ func DialPipe(path string, timeout *time.Duration) (net.Conn, error) {
 		}
 		err = waitNamedPipe(path, ms)
 		if err != nil {
```

Comment:
I think I'd have restructured this so that you *always* loop around to CreateFile.
You don't even need to check for errors at all, because CreateFile does that.

What you do need to do is bail if the time is after the absolute timeout above.  (Like 156/181).  In fact, If the time out above has expired, you should simply bail, not go down into waitNamedPipe again.

Reply:
Put another way, all waitNamedPipe() is doing for you is putting you to sleep a bit so you don't spin hard.  The consequences of its error returns are simply not very interesting.  In some other project I didn't even use this call at all, and just did a 10 ms sleep between retries...